### PR TITLE
Add Rest Client Plan Review 4.0

### PR DESCRIPTION
**Pull request by Emily-Jiang**

### CODE REVIEW

File: jwt-auth/2.1/rest-client/README.adoc [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+title: "MicroProfile Rest Client 3.1"
+date: 2024-05-15
```
Comment (WhiteCat22) [inline jwt-auth/2.1/rest-client/README.adoc:3]: Hi @Emily-Jiang how did you arrive at this date?

Everything looks good. We are just looking at the [release timeline](https://docs.google.com/spreadsheets/d/1yYI5pdRCoOT5GaZ6KMNTfg3mQ5Hwxxv45SW5WBsZAmY/edit#gid=31204409) and can't match it up to any target.

Comment (Emily-Jiang) [reply jwt-auth/2.1/rest-client/README.adoc:3]: I delayed this date a bit. Let me update to sync up with the plan to avoid confusion.

**Review by jamezp — APPROVED** [review]

**Review by WhiteCat22 — APPROVED** [review]

**Review by jamezp — APPROVED** [review]

**Comment by jclingan** [comment]
Looks good to me.

**Review by jclingan — APPROVED** [review]
Looks good.

**Pull request merged**